modules/misc/opengl.nix

```diff
@@ -0,0 +1,62 @@
+{ config, pkgs, lib, ... }:
+
+let
+  nixGL = (import (builtins.fetchTarball
+    "https://github.com/nix-community/nixGL/archive/main.tar.gz") {
+      inherit pkgs;
+    });
+
+  # Higher order function to create a function that wraps a package
+  createWrapperFunction = wrapperBinPath:
+    (
+      # Expects a PackageConfiguration hashset
+      { binPathRel, desktopFileRel, pkg, ... }:
+      # Adapted from https://nixos.wiki/wiki/Nix_Cookbook
+      pkgs.runCommand "nixgl-${pkg.pname}" {
+        buildInputs = [ pkgs.makeWrapper ];
+      } ''
+        mkdir $out
+        # Link every top-level folder from pkgs.hello to our new target
+        ln -s ${pkg}/* $out
+        # Except the bin folder
+        rm $out/bin
+        mkdir $out/bin
+        # We create the bin folder ourselves and link every binary in it
+        ln -s ${pkg}/bin/* $out/bin
+        # Except the target binary
+        rm $out/${binPathRel}
+        # Because we create this ourself, by creating a wrapper
+        makeWrapper ${wrapperBinPath} $out/${binPathRel} \
+          --add-flags ${pkg}/${binPathRel}
+      '');
+
+in {
+  meta.maintainers = with lib.maintainers; [ michaelCTS ];
+  options.workarounds.opengl = {
+    wrapperBinPath = lib.mkOption {
+      type = lib.types.str;
+      example = "$${nixGL.auto.nixGLDefault}/bin/nixGL";
+      default = "${nixGL.auto.nixGLDefault}/bin/nixGL";
+    };
+    packages = lib.mkOption {
+      type = lib.types.listOf lib.types.anything;
+      default = [ ];
+      example = ''[{
+        binPathRel = "bin/alacritty";
+        pkg = pkgs.alacritty;
+      }]'';
```

Comment:
Dunno if a custom type is needed here of it is sufficient to use `lib.types.anything`.